Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/UpdateMetadataImportDialogSequence.java

```diff
@@ -0,0 +1,216 @@
+
+    /**
+     * Open the metadata comparison dialog after a user selects a record from the hitlist dialog.
+     */
+    public void selectRecord() {
+        String recordId = Helper.getRequestParameter("ID");
+        updateMetadataDialog.setRecordIdentifier(recordId);
+
+        try {
+            LinkedList<TempProcess> processes = ServiceManager.getImportService().importProcessHierarchy(
+                recordId,
+                createProcessForm.getCurrentImportConfiguration(), 
+                createProcessForm.getProject().getId(), 
+                createProcessForm.getTemplate().getId(), 
+                1,
+                createProcessForm.getRulesetManagement().getFunctionalKeys(FunctionalMetadata.HIGHERLEVEL_IDENTIFIER)
+            );
+            createProcessForm.setProcesses(processes);
+            showMetadataComparisonDialog();
+        } catch (IOException | ProcessGenerationException | XPathExpressionException | URISyntaxException
+                     | ParserConfigurationException | UnsupportedFormatException | SAXException | DAOException
+                     | ConfigException | TransformerException | NoRecordFoundException | InvalidMetadataValueException
+                     | NoSuchMetadataFieldException e) {
+            logger.error("error when selecting record in hitlist", e);
+        }
+    }
+
+    /**
+     * Extract metadata comparison information from create process form (after catalog search was performed by the user) 
+     * and show comparison dialog.
+     */
+    private void showMetadataComparisonDialog() {
+        try {
+            if (!createProcessForm.getProcesses().isEmpty()) {
+                Process process = dataEditorForm.getProcess();
+                TempProcess tempProcess = createProcessForm.getProcesses().get(0);
+                
+                List<MetadataComparison> metadataComparisons = updateMetadataDialog.getMetadataComparisons();
+                metadataComparisons.clear();
+            
+                HashSet<Metadata> existingMetadata = getMetadata(dataEditorForm.getMetadataPanel().getLogicalMetadataRows());
+
+                RulesetManagementInterface ruleset = ServiceManager.getRulesetService().openRuleset(process.getRuleset());
+                for (Metadata metadata : tempProcess.getWorkpiece().getLogicalStructure().getMetadata()) {
+                    ProcessHelper.setMetadataDomain(metadata, ruleset);
+                }
+
+                ProcessHelper.generateAtstslFields(tempProcess, Collections.emptyList(), StringConstants.EDIT, false);
+
+                metadataComparisons.addAll(
+                    DataEditorService.initializeMetadataComparisons(
+                        process, 
+                        existingMetadata, 
+                        tempProcess.getWorkpiece().getLogicalStructure().getMetadata(), 
+                        dataEditorForm.getPriorityList(),
+                        dataEditorForm.getSelectedStructure().get().getType()
```

Comment:
Since `dataEditorForm.getSelectedStructure()` returns an `Optional<LogicalDivision>`, should we not first check whether the division `isPresent` before calling `get()` on the optional object?